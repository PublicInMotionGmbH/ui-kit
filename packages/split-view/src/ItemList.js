import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

import Item from './Item'

import { moduleName } from './config'

const propTypes = {
  className: PropTypes.string,
  itemRender: PropTypes.func,
  items: PropTypes.arrayOf(PropTypes.object).isRequired
}
const defaultProps = {
}

function ItemList (props) {
  const { className, itemRender, items, onClick, openIndex } = props
  const wrapperCls = buildClassName([moduleName, 'list'], className)

  return (
    <div className={wrapperCls}>
      {
        items.map((item, index) => (
          <Item
            key={index}
            active={openIndex === index}
            item={item}
            itemRender={itemRender}
            onClick={onClick}
          />
        ))
      }
    </div>
  )
}

ItemList.propTypes = propTypes

ItemList.defaultProps = defaultProps

export default ItemList
