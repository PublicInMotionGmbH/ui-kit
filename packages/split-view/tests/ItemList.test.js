import React from 'react'
import { shallow } from 'enzyme'

import ItemList from '../src/ItemList'
import { data } from './fixtures/mocks'

const createProps = (props = {}) => ({
  items: data,
  ...props
})
const defaultProps = createProps()
const createWrapper = (props = defaultProps) => shallow(<ItemList {...props} />)

describe('<ItemList>', () => {
  const props = createProps({ openItem: data[0] })
  let wrapper

  beforeEach(() => {
    wrapper = createWrapper(props)
  })

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render all children', () => {
    const dataLength = data.length
    expect(wrapper.children()).toHaveLength(dataLength)
  })

  it('should pass active=true prop to opened children', () => {
    const items = wrapper.find('Item')
    items.forEach((node, index) => {
      if (node.props().item === props.openItem) {
        expect(node.props().active).toBe(true)
      } else {
        expect(node.props().active).toBe(false)
      }
    })
  })
})
