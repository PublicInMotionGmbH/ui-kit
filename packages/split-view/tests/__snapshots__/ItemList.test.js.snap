// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemList> should render properly 1`] = `
<div
  className="talixo-split-view__list"
>
  <Item
    active={true}
    item={
      Object {
        "name": "Josh Smith I",
        "subtitle": "Subtitle 1",
        "title": "Title 1",
        "value": 0,
      }
    }
    itemRender={[Function]}
    key="0"
  />
  <Item
    active={false}
    item={
      Object {
        "name": "Josh Smith II",
        "subtitle": "Subtitle 2",
        "title": "Title 2",
        "value": 11,
      }
    }
    itemRender={[Function]}
    key="1"
  />
  <Item
    active={false}
    item={
      Object {
        "name": "Josh Smith III",
        "subtitle": "Subtitle 3",
        "title": "Title 3",
        "value": 22,
      }
    }
    itemRender={[Function]}
    key="2"
  />
  <Item
    active={false}
    item={
      Object {
        "name": "Josh Smith IV",
        "subtitle": "Subtitle 4",
        "title": "Title 4",
        "value": 33,
      }
    }
    itemRender={[Function]}
    key="3"
  />
  <Item
    active={false}
    item={
      Object {
        "name": "Josh Smith V",
        "subtitle": "Subtitle 5",
        "title": "Title 5",
        "value": 44,
      }
    }
    itemRender={[Function]}
    key="4"
  />
</div>
`;
